tests/e2e/scripts/init.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+# Script to install pre-requisites on vsphere node to run e2e test
+
+PATH=$PATH:/usr/local/go/bin/:/usr/local/bin
+echo "Installing Kubectl"
+curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl" && \
+sudo mv kubectl /usr/local/bin/ && \
+chmod a+x /usr/local/bin/kubectl
+
+echo "Installing jq"
+sudo apt -y install jq
+
+echo "Installing Go"
+curl -O https://dl.google.com/go/go1.16.10.linux-amd64.tar.gz
+tar xvf go1.16.10.linux-amd64.tar.gz
+sudo mv go /usr/local
+/usr/local/go/bin/go version
+go version
+
+echo "Installing Virtualbox"
+sudo apt -y update && upgrade
+dpkg -i virtualbox-6.1_6.1.26-145957~Debian~bullseye_amd64.deb
+sudo apt -y install virtualbox
+vboxmanage --version
+
+echo "Installing vagrant"
+wget https://releases.hashicorp.com/vagrant/2.2.19/vagrant_2.2.19_linux_amd64.zip
+sudo apt -y install -f unzip
+unzip vagrant_2.2.19_linux_amd64.zip
+sudo cp vagrant /usr/local/bin/
+PATH=$PATH:/usr/local/bin
+vagrant --version
+sudo apt-get -y install libarchive-tools
+vagrant plugin install vagrant-k3s
+vagrant plugin install vagrant-reload
+
+echo "Cloning repo"
+git clone https://github.com/k3s-io/k3s.git
+sudo mkdir .ssh
+sudo touch .ssh/authorized_keys
+sudo chmod 644 .ssh/authorized_keys
```

Comment:
Again, I find this sudo:ing confusing. Better run whole script with sudo or not.